plugin/pkg/scheduler/algorithmprovider/affinity/affinity.go

```diff
@@ -37,9 +37,9 @@ func affinityPredicates() util.StringSet {
 		"PodFitsResources",
 		"NoDiskConflict",
 		// Ensures that all pods within the same service are hosted on minions within the same region as defined by the "region" label
-		factory.RegisterFitPredicate("ServiceAffinity", algorithm.NewServiceAffinityPredicate(factory.PodLister, factory.ServiceLister, factory.MinionLister, []string{"region"})),
+		factory.RegisterFitPredicate("RegionAffinity", algorithm.NewServiceAffinityPredicate(factory.PodLister, factory.ServiceLister, factory.MinionLister, []string{"region"})),
```

Comment:
The current solution is flexible and uses labels to define topological levels. The levels do not themselves have any meaning within the system, and the user is free to ascribe any meaning to them. The labels could be regions/zones/racks or building/server-room/rack. 

Given that cross-region deployments is something that is not advocated, I am leaning towards simply removing the Affinity provider completely and just provide configuration examples to the user in documentation to educate them on the possible use cases.